pkg/util/taints/taints.go

```diff
@@ -202,7 +201,7 @@ func deleteTaints(taintsToRemove []v1.Taint, newTaints *[]v1.Taint) ([]error, bo
 			allErrs = append(allErrs, fmt.Errorf("taint %q not found", taintToRemove.ToString()))
 		}
 	}
-	return allErrs, removed
+	return allErrs, len(allErrs) == 0
```

Comment:
this is changing behavior, which is why the tests are failing:

previous behavior:
* `deleteTaints([b,a], &[a,c])` → `[c]`, `[taint b not found], true`
* `deleteTaints([a,b], &[a,c])` → `[c]`, `[taint b not found], false`

new behavior:
* `deleteTaints([b,a], &[a,c])` → `[c]`, `[taint b not found], false`
* `deleteTaints([a,b], &[a,c])` → `[c]`, `[taint b not found], false`

Reply:
@liggitt : which is the right behavior? Do we fix the test? (as a follow up)

@supriya-premkumar : probably better to open an issue and deal with this in another PR. for now let's `nolint` our way out of this.

Reply:
I'm… not sure. I'm not familiar with the use of the method, so I'm not quite sure what the intent is, but the different return based on what happened last is… squirrelly.

Reply:
I see. I will // nolint this one and take a second pass at doing this the right way. Thanks @liggitt and @dims 🙏🏽